File: habit-tracker/src/main/java/com/habit/habit_tracker/repository/TaskRepository.java
package com.habit.habit_tracker.repository;

import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;
import com.habit.habit_tracker.domain.Task;

@Repository
public interface TaskRepository extends JpaRepository<Task, Long> {

    @Query("SELECT t FROM Task t WHERE t.id = :taskId AND t.user.id = :userId")
    Optional<Task> findByIdAndUserId(@Param("taskId") Long taskId, @Param("userId") Long userId);

    @Query("SELECT t FROM Task t WHERE t.user.id = :userId")
    List<Task> findAllByUserId(@Param("userId") Long userId);
}

File: habit-tracker/src/main/kotlin/com/habit/habit_tracker/repository/TaskRepository.kt
package com.habit.habit_tracker.repository

import com.habit.habit_tracker.domain.Task
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface TaskRepository : JpaRepository<Task, Long> {

    @Query("SELECT t FROM Task t WHERE t.id = :taskId AND t.user.id = :userId")
    fun findByIdAndUserId(@Param("taskId") taskId: Long, @Param("userId") userId: Long): Optional<Task>

    @Query("SELECT t FROM Task t WHERE t.user.id = :userId")
    fun findAllByUserId(@Param("userId") userId: Long): List<Task>
}
